code/jobs/DoormanQueuedJobTask.php

```diff
@@ -185,6 +200,13 @@ public function canRunTask() {
 	 */
 	public function isCancelled() {
 		$this->refreshDescriptor();
-		return $this->descriptor->JobStatus === QueuedJob::STATUS_CANCELLED;
+
+		$descriptor = $this->getDescriptor();
+
+		if ($descriptor) {
+			return $this->descriptor->JobStatus === QueuedJob::STATUS_CANCELLED;
+		}
+
+		return true;
```

Comment:
This should be `return false`?

Missing descriptor doesn't really mean it was cancelled, just that it's missing?

Reply:
Changed to false